File: app/src/main/java/com/stormbreakerstudios/android/challenge/BaseActivity.java
package com.stormbreakerstudios.android.challenge;

import android.os.Build;
import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

public class BaseActivity extends AppCompatActivity {
    Navigator navigator = new Navigator();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }

    protected void toastLong(String text) {
        Toast.makeText(this, text, Toast.LENGTH_LONG).show();
    }

    protected void setTransparentStatusBarForKitKatAndAbove() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            Window w = getWindow();
            w.setFlags(
                    WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                    WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
            );
        }
    }
}

File: app/src/main/java/com/stormbreakerstudios/android/challenge/BaseActivity.kt
package com.stormbreakerstudios.android.challenge

import android.os.Build
import android.support.v7.app.AppCompatActivity
import android.view.Window
import android.view.WindowManager

abstract class BaseActivity : AppCompatActivity() {
    val navigator: Navigator by lazy(LazyThreadSafetyMode.PUBLICATION) { Navigator() }

    fun setTransparentStatusBarForKitKatAndAbove() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            val w: Window = window
            w.setFlags(
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
            )
        }
    }
}